server/app/repository/ApplicationRepository.java

```diff
@@ -90,21 +92,28 @@ private Application submitApplicationInternal(
               .eq("applicant.id", applicant.id)
               .eq("program.name", program.getProgramDefinition().adminName())
               .findList();
+
       ImmutableList<Application> drafts =
           oldApplications.stream()
               .filter(app -> app.getLifecycleStage().equals(LifecycleStage.DRAFT))
               .collect(ImmutableList.toImmutableList());
-      if (drafts.size() > 1) {
+
+      final Application application;
+      if (drafts.size() == 1) {
+        application = drafts.get(0);
+      } else if (drafts.isEmpty()) {
+        LOGGER.error(
+            "No DRAFT applications found when submitting for applicant {} program {}",
+            applicant.id,
+            program.id);
+        application = new Application(applicant, program, LifecycleStage.ACTIVE);
```

Comment:
(NB) Since this is handling creating an application object I'd consider making the log level warning and not an error.

Reply:
done